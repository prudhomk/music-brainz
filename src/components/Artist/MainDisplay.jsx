import React from 'react';
import { useState, useEffect } from 'react';
// import useArtist from '../state/useArtist';
import { artistSearch, mungeArtist } from '../../services/musicApi';
import ArtistList from '../Artist/ArtistList';

export default function MainDisplay() {
  const [offSet, setOffSet] = useState(0);
  const [page, setPage] = useState(1);
  const [artist, setArtist] = useState('');
  const [loading, setLoading] = useState(true);
  const [artistList, setArtistList] = useState([]);

  useEffect(() => {
    setLoading(false);
    return () => {
      return;
    };
  }, [handleSubmit]
  );
  
  const handleChange = ({ target }) => {
    setArtist(target.value);
  };

  const handleIncrement = async (e) => {
    setPage((prevPage) => prevPage + 1);
    setOffSet((prevOffSet) => prevOffSet + 5);
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

  const handleDecrement = async (e) => {
    setPage((prevPage) => prevPage - 1);
    setOffSet((prevOffSet) => prevOffSet - 5);
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

 
  return (
    <div data-testid="display">
      <form onSubmit={handleSubmit}>
    Artist: <input type="text" placeholder="Artist" name="searchTerm" onChange={handleChange}/>
        <button data-testid="submitButton">SEARCH</button>
      </form>
    
      <button
        data-testid="decrement"
        disabled={page <= 1}
        onClick={handleDecrement}>
                -
      </button> 
      <span data-testid="page">Page: {page}</span>
      <button 
        data-testid="increment"
        onClick={handleIncrement}>
      +
      </button>
      <ArtistList artistList={artistList}/>
    </div>
  );
}
