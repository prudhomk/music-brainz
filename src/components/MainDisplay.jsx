import React from 'react';
import useArtist from '../state/useArtist';

export default function MainDisplay() {
  const { artist, loading } = useArtist('');
  
  const handleChange = ({ target }) => {
    setSearchTerm(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
  };

  return (
    <form>
      Artist: <input type="text" placeholder="Artist" name="searchTerm" onChange={handleChange}/>
      <button>SEARCH</button>
    </form>
  );
}
