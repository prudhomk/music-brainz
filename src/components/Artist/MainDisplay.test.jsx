import React from 'react';
import { render } from '@testing-library/react';
import MainDisplay from './MainDisplay';

describe('Main display component snapshot', () => {
  it('render MainDisplay', () => {
    const { asFragment } = render(
      <MainDisplay />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
