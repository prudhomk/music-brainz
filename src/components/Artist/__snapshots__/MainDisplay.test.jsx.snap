// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main display component snapshot render MainDisplay 1`] = `
<DocumentFragment>
  <div
    data-testid="display"
  >
    <form>
      Artist: 
      <input
        name="searchTerm"
        placeholder="Artist"
        type="text"
      />
      <button
        data-testid="submitButton"
      >
        SEARCH
      </button>
    </form>
    <button
      data-testid="decrement"
      disabled=""
    >
      -
    </button>
    Page: 1
    <button
      data-testid="increment"
    >
      +
    </button>
    <ul
      data-testid="artistList"
    />
  </div>
</DocumentFragment>
`;
